File: 1833-find-the-highest-altitude/find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int [] arr1=new int[gain.length+1];
        arr1[0]=0;
        int max=0;
        for(int i=0;i<gain.length;i++){
            arr1[i+1]=arr1[i]+gain[i];
            max=Math.max(arr1[i+1],max);
        }
        return max;
    }
}

File: 1833-find-the-highest-altitude/find-the-highest-altitude.cpp
class Solution {
public:
    Solution() {
        ios::sync_with_stdio(false);
        cin.tie(nullptr);
        cout.tie(nullptr);
    }
    int largestAltitude(vector<int>& gain) {
        int ans = max(0, gain[0]);
        for(int i = 1; i < gain.size(); i++) {
            gain[i] += gain[i-1];
            ans = max(ans, gain[i]);
        }
        return ans;
    }
};